src/snapshot/index.ts

```diff
@@ -1,14 +1,16 @@
-import { createStore, Event, sample, Store } from 'effector';
+import { createStore, sample, Store, Unit } from 'effector';
+
+type NoInfer<T> = [T][T extends any ? 0 : never];
 
 export function snapshot<SourceType, TargetType = SourceType>({
   source,
   clock,
   fn = (value: SourceType) => value as unknown as TargetType,
 }: {
   source: Store<SourceType>;
-  clock?: Event<any>;
+  clock?: Unit<any>;
```

Comment:
Actually Unit is a domain and a Scope
https://github.com/effector/effector/blob/master/packages/effector/index.d.ts#L3009

what about to explicitly set `Event<T> | Store<T> | Effect<T, any, any>` ?
Otherwise we need to add tests for each kind of units (as runtime as type).

Reply:
Updated that, thank you!
I put `Unit<any>` here because it was used in similar situations in `throttle`, `spread`, `reset`, etc. - my understanding was that this is the proper way